### PR TITLE
Adding dummy structural constitutive laws

### PR DESCRIPTION
**Pull request by loumalouomega**
The name of the CL are:

- TrussConstitutiveLaw
- BeamConstitutiveLaw

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_constitutive/beam_constitutive_law.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ConstitutiveLaw::Pointer BeamConstitutiveLaw::Clone() const
+{
+    BeamConstitutiveLaw::Pointer p_clone(new BeamConstitutiveLaw(*this));
+    return p_clone;
+}
+
+//*******************************DESTRUCTOR*******************************************
+//************************************************************************************
+
+BeamConstitutiveLaw::~BeamConstitutiveLaw()
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_constitutive/beam_constitutive_law.cpp:51]: do not define the empty methods

File: applications/StructuralMechanicsApplication/custom_constitutive/beam_constitutive_law.h [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        return 1;
+    }
+
+    /**
+     * Computes the material response:
+     * PK2 stresses and algorithmic ConstitutiveMatrix
+     * @param rValues: The Internalvalues of the law
+     * @see   Parameters
+     */
+    void CalculateMaterialResponsePK2 (Parameters & rValues) override;
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_constitutive/beam_constitutive_law.h]: same

File: applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+TrussConstitutiveLaw::~TrussConstitutiveLaw()
+{
+    // TODO: Add if necessary
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void  TrussConstitutiveLaw::CalculateMaterialResponsePK2 (Parameters& rValues)
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.cpp]: same, do not redefine the empty methods

File: applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.cpp [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+//************************************************************************************
+//************************************************************************************
+
+int TrussConstitutiveLaw::Check(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+)
+{
+    if(YOUNG_MODULUS.Key() == 0 || rMaterialProperties[YOUNG_MODULUS] <= 0.00)
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.cpp:77]: the check shouls check the proper variables

File: applications/StructuralMechanicsApplication/structural_mechanics_application.cpp [hunk 5]
```diff
@@ -181,10 +178,10 @@ void KratosStructuralMechanicsApplication::Register()
     KRATOS_REGISTER_VARIABLE( MEAN_RADIUS )
     KRATOS_REGISTER_VARIABLE( SECTION_SIDES )
     KRATOS_REGISTER_VARIABLE( GEOMETRIC_STIFFNESS )
-	KRATOS_REGISTER_VARIABLE(LOCAL_AXIS_VECTOR_1)
-	KRATOS_REGISTER_VARIABLE(LOCAL_ELEMENT_ORIENTATION)
-	KRATOS_REGISTER_VARIABLE(ORTHOTROPIC_ORIENTATION_ASSIGNMENT)
-	KRATOS_REGISTER_VARIABLE(ORTHOTROPIC_FIBER_ORIENTATION_1)
+    KRATOS_REGISTER_VARIABLE(LOCAL_AXIS_VECTOR_1)
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/structural_mechanics_application.cpp:181]: do not define LOCAL_AXIS_VECTOR_1

use LOCAL_AXIS_1 instead.

but mostly...why are u adding these variables or touching this file in this PR??

File: applications/StructuralMechanicsApplication/structural_mechanics_application.h [hunk 6]
```diff
@@ -69,6 +69,8 @@
 #include "custom_conditions/surface_load_condition_3d.h"
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/structural_mechanics_application.h:69]: should not be a part of this PR

File: applications/StructuralMechanicsApplication/structural_mechanics_application_variables.h [hunk 7]
```diff
@@ -52,10 +52,10 @@ typedef std::vector<MpcDataPointerType>
     KRATOS_DEFINE_APPLICATION_VARIABLE( STRUCTURAL_MECHANICS_APPLICATION, double, MEAN_RADIUS )
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/structural_mechanics_application_variables.h:52]: should not be a part of this pr

File: applications/MeshingApplication/custom_processes/metrics_levelset_process.h [hunk 8]
```diff
@@ -248,7 +248,10 @@ class ComputeLevelSetSolMetricProcess
     ///@{
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/metrics_levelset_process.h:248]: should not be a part of this pr

File: applications/StructuralMechanicsApplication/custom_conditions/line_load_condition_2d.cpp [hunk 9]
```diff
@@ -285,7 +285,6 @@ namespace Kratos
         )
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_conditions/line_load_condition_2d.cpp:285]: should not be a part of this pr

File: applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.h [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+        return 1;
+    }
+
+    /**
+     * Computes the material response:
+     * PK2 stresses and algorithmic ConstitutiveMatrix
+     * @param rValues: The Internalvalues of the law
+     * @see   Parameters
+     */
+    void CalculateMaterialResponsePK2 (Parameters & rValues) override;
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_constitutive/truss_constitutive_law.h]: same

**Review by jginternational — APPROVED** [review]
Like

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
do not forget to remove the files that should not be there

**Comment by RiccardoRossi** [comment]
i think it would not be bad to merge this for the release.

@pooyan-dadvand if you disagree just remove the Release tag

**Comment by RiccardoRossi** [comment]
sorry i put the comment in the wrong place. This definitely needs to go in the release since it is needed for the Beam and truss GUI interface

**Review by RiccardoRossi — APPROVED** [review]
ok as we spoke i am accepting this under the understanding that you remove the repeated methods.

**Pull request merged**